From af22063ebf3e4da9cf6c7e793dfc4a502b2d230c Mon Sep 17 00:00:00 2001
From: Thomas Bechtold <tbechtold@suse.com>
Date: Tue, 4 Jul 2017 12:26:39 +0200
Subject: [PATCH] Allow using openstackdocstheme without git installed

When building rpm packages, git might not be available in the
build env because it is not needed when building from a sdist tarball.
So make git optional.

Change-Id: I159768799fcf15ed0aea6b8350dd395f632e812d
---
 api-ref/source/conf.py         |  8 ++++++--
 openstackdocstheme/__init__.py | 11 ++++++++---
 2 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/api-ref/source/conf.py b/api-ref/source/conf.py
index 66a4865..e4a4ee6 100644
--- a/api-ref/source/conf.py
+++ b/api-ref/source/conf.py
@@ -71,8 +71,12 @@ release = '1.0'
 # These variables are passed to the logabug code via html_context.
 giturl = u'https://git.openstack.org/cgit/openstack/openstackdocstheme/tree/doc/source'
 git_cmd = ["/usr/bin/git", "rev-parse", "HEAD"]
-gitsha = subprocess.Popen(
-    git_cmd, stdout=subprocess.PIPE).communicate()[0].strip('\n')
+# git might not be available during build (eg when building from an sdist)
+try:
+    gitsha = subprocess.Popen(
+        git_cmd, stdout=subprocess.PIPE).communicate()[0].strip('\n')
+except Exception:
+    gitsha = 'unknown'
 bug_tag = "doc-builds"
 
 html_context = {"gitsha": gitsha, "bug_tag": bug_tag, "giturl": giturl}
diff --git a/openstackdocstheme/__init__.py b/openstackdocstheme/__init__.py
index 08f29f8..16282f7 100644
--- a/openstackdocstheme/__init__.py
+++ b/openstackdocstheme/__init__.py
@@ -78,9 +78,14 @@ def _html_page_context(app, pagename, templatename, context, doctree):
     global _html_context_data
     if _html_context_data is None:
         _html_context_data = {}
-        _html_context_data['gitsha'] = subprocess.check_output(
-            ['git', 'rev-parse', 'HEAD'],
-        ).decode('utf-8').strip()
+        try:
+            _html_context_data['gitsha'] = subprocess.check_output(
+                ['git', 'rev-parse', 'HEAD'],
+            ).decode('utf-8').strip()
+        except Exception:
+            app.warn('Cannot get gitsha from git repository.')
+            _html_context_data['gitsha'] = 'unknown'
+
         repo_name = app.config.repository_name
         if repo_name:
             _html_context_data['giturl'] = _giturl.format(repo_name)
-- 
2.13.2

